tensorflow/tools/ci_build/ci_parameterized_build.sh

```diff
@@ -120,15 +128,20 @@ fi
 EXTRA_PARAMS=""
 
 # Determine if Docker is available
-MAIN_CMD=${DOCKER_MAIN_CMD}
 if [[ -z "$(which docker)" ]]; then
+  DO_DOCKER=0
+
   echo "It appears that Docker is not available on this system. "\
 "Will perform build without Docker."
-  echo "In addition, the additional option flags will be applied to the build:"
+  echo "Also, the additional option flags will be applied to the build:"
   echo "  ${NO_DOCKER_OPT_FLAG}"
-  MAIN_CMD=${NO_DOCKER_MAIN_CMD}
+  MAIN_CMD="${NO_DOCKER_MAIN_CMD} ${CTYPE}"
   OPT_FLAG="${OPT_FLAG} ${NO_DOCKER_OPT_FLAG}"
 
+  # Perform bazel clean in non-Docker environments to ensure a clean build
+  echo "Also, \"${BAZEL_CLEAN_CMD}\" will be performed prior to "\
+"the main build step"
+  DO_BAZEL_CLEAN=1
```

Comment:
@jendap Yes, that would slow things down. But I see your point. I guess the Bazel caching somehow didn't work when the folder is mapped from outside Docker to inside, which is the main reason why we haven't needed to do bazel clean for Docker builds? @martinwicke has their been any changes regarding the caching recently?